sdk/core/Azure.Core/tests/public/PatchModels/ChildPatchModel.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Json;
+
+namespace Azure.Core.Tests.PatchModels
+{
+    /// <summary>
+    /// This model illustrates a nested child model in a parent model.
+    /// </summary>
+    public partial class ChildPatchModel
+    {
+#pragma warning disable AZC0020 // Avoid using banned types in libraries
+        private readonly MutableJsonElement _element;
+
+        // Note: A child patch model doesn't have a public constructor.
```

Comment:
We have public constructor currently. Is there any reason we change this?

Reply:
My thinking is that, for Patch models, we need a way to connect the MutableJsonElement in the child model to the MutableJsonElement in the parent model, so that they point to the same root MutableJsonDocument.  Without this, they don't access the same changelist and writing the Patch JSON would be a lot more complicated.

We could make it so that nested models have a public constructor but can't be set on the parent (shown [here](https://github.com/Azure/azure-sdk-for-net/pull/38159/files#diff-d65d3d083db0614045d2be9cde55cfbf9227c0e5de7d0a93d96ea9d54ec3a684R81-R83)), but if you can't set the child model, then I don't know why you would have a public constructor on the child model - callers could create an instance of it but then not do anything with it, which seems like it could be frustrating to users.

There may be times when a nested model is also an input model on a service method - when that happens the model will need to have a public constructor, and we will need to make the parent model not settable so we can connect the MutableJsonElements.  It would be ideal if the generator could detect that case and generate the model with the public constructor when that happens.